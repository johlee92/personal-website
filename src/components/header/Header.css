.header-container {
    flex-shrink: 0;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    padding: 10px 0;
    align-items: center;
    border-bottom: darkgray 2px solid;
}

.header-container > p {
    margin: 0;
    font-size: 18px;
    font-weight: bold;
}

.header-image {
    width: 25px;
    height: 25px;
}