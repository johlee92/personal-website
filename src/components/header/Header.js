import React, { Component } from 'react';
import './Header.css';
import logo from '../../images/logo.png'
import menuPic from '../../images/menuHamburger.png';

class Header extends Component {
    render() {
        return (
            <div className="header-container">
                <img src={logo} alt="JYL logo" className="header-image"/>
                <p>John</p>
                <img src={menuPic} alt="JYL logo" className="header-image"/>
            </div>
        )
    }
}

export default Header;