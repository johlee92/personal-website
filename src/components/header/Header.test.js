import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe(`Header component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
                <Header />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <Header />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
