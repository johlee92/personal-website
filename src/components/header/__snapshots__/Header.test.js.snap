// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the component as expected 1`] = `
<div
  className="header-container"
>
  <img
    alt="JYL logo"
    className="header-image"
    src="logo.png"
  />
  <p>
    John
  </p>
  <img
    alt="JYL logo"
    className="header-image"
    src="menuHamburger.png"
  />
</div>
`;
